### ISSUE TITLE
rabbitmq_user: First time creating user always fails

### ISSUE THREAD

**Issue by bradwilson** [ansible/ansible-modules-extras#1671]
##### Issue Type:

Bug Report
##### Ansible Version:

Running ansible from source, commit @6ddea3e915012a381b53e34b19c2270cf5e4e815
##### Ansible Configuration:

There is no /etc/ansible/ansible.cfg file.
##### Environment:

Ubuntu 14.04 (issue presumably not environment dependent)
##### Summary:

The first time creating a user will always fail, because the default value for `self._permissions` is `None` instead of `[]`. When the system attempts to retrieve the existing permissions, it (correctly) does not find any permissions, which causes `self._permissions` to remain untouched. When the permission sets are being compared, the code fails with `'NoneType' is not iterable`.
##### Steps To Reproduce:

Use `rabbitmq_user` to create a new user. Permissions are not necessary for the repro; simply providing a username and password is sufficient.
##### Expected Results:

User is created with correct values, including permissions.
##### Actual Results:

The user is created, but the playbook fails when attempting to determine what to do about permissions (regardless of whether they are empty or not):

```
Traceback (most recent call last):
  File \"/root/.ansible/tmp/ansible-tmp-1455576017.44-86871992316534/rabbitmq_user\", line 2348, in <module>
    main()
  File \"/root/.ansible/tmp/ansible-tmp-1455576017.44-86871992316534/rabbitmq_user\", line 294, in main
    rabbitmq_user.set_permissions()
  File \"/root/.ansible/tmp/ansible-tmp-1455576017.44-86871992316534/rabbitmq_user\", line 205, in set_permissions
    for permission in self._permissions:
TypeError: 'NoneType' object is not iterable
```

**Comment by flaviogrossi** [comment]
i think this should be fixed in pullreqs #1656 and and #1657

**Comment by bradwilson** [comment]
Duplicate.
